lib/Dialect/LLHD/IR/LLHDOps.cpp

```diff
@@ -221,6 +221,56 @@ OpFoldResult llhd::SModOp::fold(ArrayRef<Attribute> operands) {
   });
 }
 
+//===----------------------------------------------------------------------===//
+// EqOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult llhd::EqOp::fold(ArrayRef<Attribute> operands) {
+  /// llhd.eq(x, 1) -> x
+  if (matchPattern(rhs(), m_One()) && lhs().getType().isSignlessInteger(1))
+    return lhs();
+
+  /// llhs.eq(x,x) -> 1
+  if (lhs() == rhs())
+    return BoolAttr::get(true, getContext());
+
+  if (!operands[0] || !operands[1])
+    return {};
+
+  if (auto lhs = operands[0].dyn_cast<IntegerAttr>()) {
+    if (auto rhs = operands[1].dyn_cast<IntegerAttr>()) {
+      return BoolAttr::get(lhs.getValue() == rhs.getValue(), getContext());
```

Comment:
nit: Attributes are unique'd, so you should be able to check pointer equality here and below.